### DISCUSSION TITLE
URL Connection RustDesk

### DISCUSSION THREAD

**Discussion by AaronHuston** [rustdesk/rustdesk#5299] [Q&A] [answered]
Hello, before upgrading to version 1.2.1, I connected to the client via url. Now, when I click on the url link, it sends the entire string to rustdesk, and not just the client ID

For example, writing in the browser `rustdesk://12345678` ran rustdesk and connected to the client

Has something changed? Can you tell me what to do to connect as before?

Line in the registry:
`"C:\Program Files\RustDesk\RustDesk.exe"  --play "%0"`

**Comment by dinger1986** [comment]
presumably you are on windows? It works fine for me, I even just tested it there

**Comment by AaronHuston** [comment]
Yes, on windows, it just passes the whole line to rustdesk, I can't understand why it happened

![image](https://github.com/rustdesk/rustdesk/assets/111519038/2040d867-4def-46ff-b0cb-ba686bbe8b6d)

**Reply by dinger1986** [reply]
how are you calling it? I use it via a url action on an rmm and works fine

**Reply by AaronHuston** [reply, marked as answer]
It works for me by passing an argument through the windows registry

Now I understand why it doesn't work. Need to rewrite the line to `"C:\Program Files\RustDesk\RustDesk.exe" %1"`

We write `rustdesk://client_Id` in the browser line and pass the argument to the program launch
Ready registry branch for working with url: 
```
Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\rustdesk]
@="URL:cmd Protocol"
"URL Protocol"=""

[HKEY_CLASSES_ROOT\rustdesk\shell]
@=""

[HKEY_CLASSES_ROOT\rustdesk\shell\open]
@=""

[HKEY_CLASSES_ROOT\rustdesk\shell\open\command]
@="\"C:\\Program Files\\RustDesk\\RustDesk.exe\"  %1\""

```

**Reply by dinger1986** [reply]
but it should have already been there

**Comment by william-bohannan** [comment]
If you create a desktop shortcut of a connection in Rustdesk and look at the created shortcut properties, it is something like this:

"C:\Program Files\RustDesk\rustdesk.exe" --connect 78581911

So i create a new rustdesk-patch.reg that has the updated format (after installing rustdesk), tested OK on Windows 11.

```
Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\.rustdesk\shell\open\command]
@="\"C:\\Program Files\\RustDesk\\RustDesk.exe\" --connect \"%1\""
```

Now accepts from edge
```
rustdesk://78581911
```

**Comment by celevra** [comment]
is there a way to also set the connection password in the url?

**Reply by adamsthws** [reply]
> is there a way to also set the connection password in the url?

A little late with this but leaving it here for any future visitors...

To start a Rustdesk session from url including both the ID and password, use this:

```

rustdesk://connection/new/<rustdesk-id>?password=<rustdesk-pw>
# eg:
rustdesk://connection/new/1123123123?password=Password123

```
